petab/sbml.py

```diff
@@ -98,3 +98,199 @@ def constant_species_to_parameters(sbml_model):
             reaction.removeProduct(speciesId)
 
     return transformables
+
+
+def is_sbml_consistent(sbml_document: libsbml.SBMLDocument, check_units=False):
+    """Check for SBML validity / consistency
+
+    Arguments:
+        sbml_document: SBML document to check
+        check_units: Also check for unit-related issues
+    Returns:
+        False if problems were detected, otherwise True
+    """
+
+    if not check_units:
+        sbml_document.setConsistencyChecks(
+            libsbml.LIBSBML_CAT_UNITS_CONSISTENCY, False)
+
+    has_problems = sbml_document.checkConsistency()
+    if has_problems:
+        print_sbml_errors(sbml_document)
+        print('WARNING: Generated invalid SBML model. Check messages above.')
```

Comment:
logger.warn? By default gives the same output, and is easier to write e.g. to file + screen.

I usually do

```python
import logging

logger = logging.getLogger(__name__)
```

Reply:
same for all other print commands.